website/conferences/views.py

```diff
@@ -221,8 +331,54 @@ def serialize_conference(conf):
         'talk': conf.talk,
     }
 
+def get_meetings_page():
+    """
+    Returns requested page number, fetched from query param.
+    If none, returns 1 by default.
+    """
+    if request.args.get('page', None):
+        try:
+            return int(request.args.get('page'))
+        except ValueError:
+            return 1
+    return 1
+
+def create_pagination_array(current_page, total_pages):
+    """
+    Returns an array of page numbers for pagination in UI. Creates similar UI
+    to ember addon https://github.com/mharris717/ember-cli-pagination
+
+    For example: [1, 2, '...', 23, 24, 25, 26], where current page is 25.
+    Or, [1, 2, '...', 6, 7, 8, 9, 10, '...', 25, 26], where current page is 8.
+    : param integer current_page: current page of results being displayed
+    : param integer total_pages: total number of pages available
+    """
+    if total_pages <= 4:
+        return range(1, total_pages + 1)
+
+    # Pagination available will always include the first two and last two pages
+    start = [1, 2]
+    end = [total_pages - 1, total_pages]
+
+    # Middle range of pagination available will take the form:
+    # [current - 2, current - 1, current, current + 1, current + 2]
+    middle = []
+    for page_num in range(current_page - 2, current_page + 3):
+        if page_num > 2 and page_num < total_pages - 1:
+            middle.append(page_num)
 
-def conference_results(meeting):
+    # If there is a gap between the start range and the middle range, add a '...'
+    if not middle or middle[0] != 3:
```

Comment:
The code for displaying pages to show (including the ellipsis) should probably happen on the frontend since it's presentational logic (rather than "business" logic).